### PR TITLE
Feat (API) Add Amqp Struct and add it to router

### PR DESCRIPTION
**Pull request by Isilud**
Depends on [#352 ](https://github.com/sogilis/Voogle/pull/352)

### CODE REVIEW

File: src/cmd/api/controllers/websocket_handler.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	log "github.com/sirupsen/logrus"
+)
+
+type WSHandler struct {
+}
+
+func (wsh WSHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+
+	upgrader := websocket.Upgrader{}
+	upgrader.CheckOrigin = func(r *http.Request) bool { return true }
```
Comment (vlamy) [inline src/cmd/api/controllers/websocket_handler.go:16]: Is it normal that we bypass origin checking ?

Comment (vlamy) [inline src/cmd/api/controllers/websocket_handler.go:16]: Is it normal that we bypass origin checking ?

File: src/pkg/clients/Amqp.go [hunk 2]
```diff
@@ -100,3 +100,131 @@ func (r *amqpClient) connect() error {
...
+var _ IAmqpExchanger = &amqpExchanger{}
+
+type amqpExchanger struct {
+	channel       *amqp.Channel
+	fullAddr      string
+	exchangerName string
+}
+
+func NewAmqpExchanger(addr, user, pwd, exchangerName string) (IAmqpExchanger, error) {
+	amqpE := &amqpExchanger{
```
Comment (RaboliotLeGris) [inline src/pkg/clients/Amqp.go:120]: I don't think amqpE is really meaningful in this case

File: src/pkg/clients/Amqp.go [hunk 3]
```diff
@@ -100,3 +100,131 @@ func (r *amqpClient) connect() error {
...
+		// 	return err
+		// }
+
+		return r.channel.Publish(
+			r.exchangerName,
+			key,
+			false,
+			false,
+			amqp.Publishing{
+				ContentType: "text/plain",
```
Comment (RaboliotLeGris) [inline src/pkg/clients/Amqp.go:190]: I saw in the futur (or the next PR) that we use json for our publication message and it should be protobuf.

We should only use JSON for our communication front <-> API. For all the other cases, it should be protobuf.

**Pull request merged**